todo/src/domain/todo.rs

```diff
@@ -1,26 +1,30 @@
 use std::fmt;
 
+use uuid::Uuid;
+
 #[derive(Debug, Clone, PartialEq)]
 pub struct Todo {
     pub message: String,
     pub done: bool,
+    pub id: Option<Uuid>,
 }
 
 impl Todo {
     pub fn new(message: String) -> Self {
         Todo {
             message,
             done: false,
+            id: None,
         }
     }
 }
 
 impl fmt::Display for Todo {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         if !self.done {
-            write!(f, "[ ] - {}", self.message)?
+            write!(f, "[ ] - {} - id: {}", self.message, self.id.unwrap())?
         } else {
-            write!(f, "[X] - {}", self.message)?
+            write!(f, "[X] - {} - id: {}", self.message, self.id.unwrap())?
         }
```

Comment:
You could extract the `marker_char`, e.g.

```rs
let marker = if self.done { 'X' } else { ' ' };
```

Reply:
nice catch